src/ts/deploy.ts

```diff
@@ -36,4 +37,38 @@ export type DeploymentArguments<
   ? [string]
   : T extends typeof CONTRACT_NAMES.settlement
   ? [string]
-  : never;
+  : unknown[];
```

Comment:
Why is this change needed?

Reply:
So the issue is that `artifacts.readArtifact` returns an artifact without a name known at compile time. In those cases, the type of the deployment arguments are `unknown`.

I couldn't think of a better solution, so I ended up setting up the types so:
```
const artifact: Artifact = artifacts.readArtifact("GPv2Settlement");
const { abi, bytecode } = artifact;

deterministicDeploymentAddress({ contractName: "GPv2Settlement", abi, bytecode }, [someAddress]) // OK
deterministicDeploymentAddress(artifact, [someAddress]) // OK

deterministicDeploymentAddress({ contractName: "whatever", abi, bytecode }, []) // COMPILE error, not a NamedArtifact<C> | Artifact
deterministicDeploymentAddress({ contractName: "GPv2Settlement", abi, bytecode }, [42]) // COMPILE error, expected a string

deterministicDeploymentAddress({ contractName: "GPv2Settlement", abi, bytecode }, ["notaddress"]) // RUNTIME error, expected an address
deterministicDeploymentAddress(artifact, [42]) // RUNTIME error, expected an address
deterministicDeploymentAddress(artifact, ["notaddress"]) // RUNTIME error, expected an address
```

Basically, `DeploymentArguments` has a well-defined type for a contract name known at compile-time, and `unknown[]` for one that isn't.